pytest/tests/loadtest/locust/common/social.py

```diff
@@ -0,0 +1,279 @@
+
+
+class InitSocialDbAccount(Transaction):
+    """
+    Send initial storage balance to ensure the account can use social DB.
+
+    Technically, we could also rely on lazy initialization and just send enough
+    balance with each request. But a typical user sends balance ahead of time.
+    """
+
+    def __init__(self, contract_id: str, account: Account):
+        super().__init__()
+        self.contract_id = contract_id
+        self.account = account
+
+    def sign_and_serialize(self, block_hash):
+        args = json.dumps({"account_id": self.account.key.account_id})
+        return transaction.sign_function_call_tx(self.account.key,
+                                                 self.contract_id,
+                                                 "storage_deposit",
+                                                 args.encode('utf-8'),
+                                                 300 * TGAS, 1 * NEAR_BASE,
+                                                 self.account.use_nonce(),
+                                                 block_hash)
+
+
+def social_db_build_index_obj(key_list_pairs: dict) -> dict:
+    """
+    JSON serializes the key - value-list pairs to be included in a SocialDB set message.
+
+    To elaborate a bit more, SocialDB expects for example
+    ```json
+    "index": { "graph": value_string } 
+    ```
+    where value_string = 
+    ```json
+    "[{\"key\":\"follow\",\"value\":{\"type\":\"follow\",\"accountId\":\"pagodaplatform.near\"}}]"
+    ```
+    So it's really JSON nested inside a JSON string.
+    And worse, the nested JSON is always a list of objects with "key" and "value" fields.
+    This method unfolds this format from a leaner definition, using a dict to
+    define each `value_string`.
+    """
+    obj = {}
+    for index_key, values in key_list_pairs.items():
+        unfolded_list = [{"key": k, "value": v} for k, v in values.items()]
+        obj[index_key] = json.dumps(unfolded_list)
+    return obj
+
+
+def social_db_set_msg(sender: str, values: dict, index: dict) -> str:
+    """
+    Construct a SocialDB `set` function argument.
+
+    The output will be of the form:
+    ```json
+    {
+      "data": {
+        "key1": value1,
+        "key3": value2,
+        "key4": value3,
+        "index": {
+          "index_key1": "[{\"index_key_1_key_A\":\"index_key_1_value_A\"}]",
+          "index_key2": "[{\"index_key_2_key_A\":\"index_key_2_value_A\"},{\"index_key_2_key_B\":\"index_key_2_value_B\"}]",
+        }
+      }
+    }
+    ```
+    """
+    updates = values
+    updates["index"] = social_db_build_index_obj(index)
+    msg = {"data": {sender: updates}}
+    return json.dumps(msg)
+
+
+def social_follow_args(sender: str, follow_list: "list[str]"):
+    follow_map = {}
+    graph_map = {"follow": {}}
+    notify_map = {}
+    for user in follow_list:
+        follow_map[user] = ""
+        graph_map["follow"] = {"type": "follow", "accountId": user}
```

Comment:
Note, that `graph_map["follow"]` will be overwritten multiple times in this loop, at the end containing only the record for the last user in the `follow_list`. Is this intended or should we merge info across users?